modules/core/src/smithy4s/Hints.scala

```diff
@@ -170,19 +176,81 @@ object Hints {
   }
 
   object Binding {
-    final case class StaticBinding[A](key: ShapeTag[A], value: A)
-        extends Binding {
+    final case class StaticBinding[A](
```

Comment:
Do you think we could make it a normal class ?

Reply:
Probably, but do we want to? We'll need to make a Mirror for 3.x.

Reply:
right, let's not then. I do want to get rid of most publicly-facing case classes at some point but we'll keep that for a breaking release, provided we can get some cross-version tooling for macro-based datatypes.